compiler/compiler/src/lib.rs

```diff
@@ -77,13 +79,11 @@ impl<'a> Compiler<'a> {
     }
 
     // Parses and stores a program file content from a string, constructs a syntax tree, and generates a program.
-    pub fn parse_program_from_string(&mut self, program_string: &str) -> Result<()> {
+    pub fn parse_program_from_string(&mut self, program_string: &str, name: FileName) -> Result<()> {
+        let prg_sf = with_session_globals(|s| s.source_map.new_source(program_string, name));
```

Comment:
It's not clear to me what `prg_sf` stands for. Program source file?

Could you add a comment describing what this line does. I assume it sets up the error handler and we should document that.